common/djangoapps/third_party_auth/api/urls.py

```diff
@@ -20,4 +20,9 @@
         UserMappingView.as_view(),
         name='third_party_auth_user_mapping_api',
     ),
+    url(
+        r'^v0/providers/user_status$',
```

Comment:
@douglashall Does the API added here relate to the `users` resource rather than the `providers` resource?  I believe the `providers` resource is reserved for endpoints that are provider-facing.

Reply:
No, this API endpoint is associated with the `providers` resource. We are basically returning a list of the available TPA providers configured in the system with the currently logged-in user's "linked" status with respect to each of them.

Reply:
To clarify, the `providers` resource is intended to be used by providers for server-to-server calls.  See [the doscstring for the Providers view](https://github.com/edx/edx-platform/blob/712805992193da49e7ff72a0b396334c7494cac5/common/djangoapps/third_party_auth/api/views.py#L267).  Whereas the `users` resource is intended for access by users to retrieve information for each provider.  See [the docstring for the Users view](https://github.com/edx/edx-platform/blob/712805992193da49e7ff72a0b396334c7494cac5/common/djangoapps/third_party_auth/api/views.py#L204).  In fact, it's also now unclear how this new view differs from the pre-existing `Users` view, since they provide similar data, with the new endpoint just being a superset of the old one.

Reply:
In my experience with REST-ful APIs, the resource doesn't have anything to do with who is consuming the API endpoint. The resource is meant to describe the entity for which data is being returned/updated. Perhaps it would have been better to keep this closer to the user account api as I originally had it. I think what we have here is good enough. I would be happy to review your PR if you have a better idea about how this code should be written.